components/help/components/formatting.jsx

```diff
@@ -5,81 +5,414 @@ import React from 'react';
+                />
+            </p>
+            <p>
+                <FormattedMessage
+                    id='help.formatting.example'
+                    defaultMessage='Example:'
+                />
+            </p>
+            <FormattedMessage
+                id='help.formatting.codeBlock'
+                defaultMessage='Code block'
+            >
+                {(example) => renderRawExampleWithResult('```\n' + example + '\n```')}
+            </FormattedMessage>
+            <h3 className='markdown__heading'>
+                <FormattedMessage
+                    id='help.formatting.syntax.title'
+                    defaultMessage='Syntax Highlighting'
+                />
+            </h3>
+            <p>
+                <FormattedMarkdownMessage
+                    id='help.formatting.syntax.description'
+                    defaultMessage='To add syntax highlighting, type the language to be highlighted after the ``` at the beginning of the code block. Mattermost also offers four different code themes (GitHub, Solarized Dark, Solarized Light, Monokai) that can be changed in **Account Settings** > **Display** > **Theme** > **Custom Theme** > **Center Channel Styles**'
+                />
+            </p>
+            <p>
+                <FormattedMarkdownMessage
+                    id='help.formatting.supportedSyntax'
+                    defaultMessage={'Supported languages are: `as`, `applescript`, `osascript`, `scpt`, `bash`, `sh`, `zsh`, `clj`, `boot`, `cl2`, `cljc`, `cljs`, `cljs.hl`, `cljscm`, `cljx`, `hic`, `coffee`, `_coffee`, `cake`, `cjsx`, `cson`, `iced`, `cpp`, `c`, `cc`, `h`, `c++`, `h++`, `hpp`, `cs`, `csharp`, `css`, `d`, `di`, `dart`, `delphi`, `dpr`, `dfm`, `pas`, `pascal`, `freepascal`, `lazarus`, `lpr`, `lfm`, `diff`, `django`, `jinja`, `dockerfile`, `docker`, `erl`, `f90`, `f95`, `fsharp`, `fs`, `gcode`, `nc`, `go`, `groovy`, `handlebars`, `hbs`, `html.hbs`, `html.handlebars`, `hs`, `hx`, `java`, `jsp`, `js`, `jsx`, `json`, `jl`, `kt`, `ktm`, `kts`, `less`, `lisp`, `lua`, `mk`, `mak`, `md`, `mkdown`, `mkd`, `matlab`, `m`, `mm`, `objc`, `obj-c`, `ml`, `perl`, `pl`, `php`, `php3`, `php4`, `php5`, `php6`, `ps`, `ps1`, `pp`, `py`, `gyp`, `r`, `ruby`, `rb`, `gemspec`, `podspec`, `thor`, `irb`, `rs`, `scala`, `scm`, `sld`, `scss`, `st`, `styl`, `sql`, `swift`, `tex`, `vbnet`, `vb`, `bas`, `vbs`, `v`, `veo`, `xml`, `html`, `xhtml`, `rss`, `atom`, `xsl`, `plist`, `yaml`'}
+                />
+            </p>
+            <p>
+                <FormattedMessage
+                    id='help.formatting.example'
+                    defaultMessage='Example:'
+                />
+            </p>
+            <FormattedMessage
+                id='help.formatting.syntaxEx'
+                defaultMessage={'```go\npackage main\nimport "fmt"\nfunc main() \\{\n    fmt.Println("Hello, 世界")\n\\}\n```'}
```

Comment:
Escaping `}` and `{` with `\\` is causing to render `\}` and `\{`, respectively.

![screen shot 2019-01-17 at 12 01 29 am](https://user-images.githubusercontent.com/5334504/51261770-e0246000-19eb-11e9-843a-21bfbc7d1e3a.png)

Above rendering is not consistent among OS platforms, and it may be related to `react-intl`.  One approach is to use unicode character when escaping.  So it could be like:

```suggestion
                defaultMessage={'```go\npackage main\nimport "fmt"\nfunc main() \u29F5{\n    fmt.Println("Hello, 世界")\n\u29F5}\n```'}
```

Reply:
You're right, this is a bug in `react-intl`: https://github.com/yahoo/react-intl/issues/1158. Unfortunately, the approach with `u29F5` produces an error in my browser:
```
[React Intl] Error formatting the default message for: "help.formatting.syntaxEx"
SyntaxError: Expected ",", "}" or [^ \t\n\r,.+={}#] but "." found.
```
I found this workaround https://github.com/osmlab/maproulette3/pull/598 to solve the issue and I used it in our case:
```JSX
<FormattedMessage
    id='help.formatting.syntaxEx'
    defaultMessage={'```go\npackage main\nimport "fmt"\nfunc main() \\{\n    fmt.Println("Hello, 世界")\n\\}\n```'}
    values={{dummy: ''}}
>
    {(example) => renderRawExample(example)}
</FormattedMessage>
```
Now it should render correctly in production mode.